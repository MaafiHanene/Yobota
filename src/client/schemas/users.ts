import gql from 'graphql-tag';

export const USERS_QUERY = gql`
	query Users($lang: LanguageType!) {
		users(lang: $lang) {
			id
			first_name
			last_name
			email
			industry
			years_of_experience
			salary
			date_of_birth
		}
	}
`;
