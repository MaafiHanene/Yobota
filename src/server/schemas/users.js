import {
  GraphQLEnumType,
  GraphQLInt,
  GraphQLList,
  GraphQLObjectType,
  GraphQLString,
  GraphQLFloat,
} from 'graphql';
import dummyData from '../data/data';

// import axios from 'axios';

const UserType = new GraphQLObjectType({
  name: 'Users',
  fields: {
    id: { type: GraphQLString },
    first_name: { type: GraphQLString },
    last_name: { type: GraphQLString },
    email: { type: GraphQLString },
    industry: { type: GraphQLString },
    years_of_experience: { type: GraphQLFloat },
    salary: { type: GraphQLFloat },
    date_of_birth: { type: GraphQLString },
  },
});

// const baseURL = process.env.BACKEND_SERVICES_URL;

const activityResolver = async ({ type, id, lang }) => {
  // let result = await axios.get(`${baseURL}/user?lang=${lng}`);
  // const lng = lang || 'en';
  let result = dummyData;
  if (id) {
    result = result.filter((activity) => activity.id === id);
    return result[0];
  }
  if (type !== undefined) {
    result = result.filter((activity) => activity.type === type);
  }
  return result;
};

const LanguageEnumType = new GraphQLEnumType({
  name: 'LanguageType',
  values: {
    EN: { value: 'en' },
    FR: { value: 'fr' },
    ES: { value: 'es' },
  },
});

const UserSchema = {
  type: UserType,
  args: {
    id: { type: GraphQLInt },
    lang: { type: LanguageEnumType },
  },
  // eslint-disable-next-line no-undef
  resolve: (_, { id, lang }) => activityResolver({ id, lang }),
};

const UsersSchema = {
  type: new GraphQLList(UserType),
  args: {
    // type: { type: ActivityEnumType },
    lang: { type: LanguageEnumType },
  },
  // eslint-disable-next-line no-undef
  resolve: (_, { type, lang }) => activityResolver({ type, lang }),
};

export {
  activityResolver,
  UserType,
  LanguageEnumType,
  UserSchema,
  UsersSchema,
};
