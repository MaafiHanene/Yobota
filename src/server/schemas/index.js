import { GraphQLObjectType, GraphQLSchema } from 'graphql';
import { UserSchema, UsersSchema, /* UserMutationSchema */ } from './users';

const schema = new GraphQLSchema({
  query: new GraphQLObjectType({
    name: 'RootQueryType',
    fields: {
      user: UserSchema,
      users: UsersSchema,
    },
  }),
  // mutation: new GraphQLObjectType({
  //   name: 'RootMutationType',
  //   fields: {
  //     newUsers: UserMutationSchema,
  //   },
  // }),
});

export default schema;
