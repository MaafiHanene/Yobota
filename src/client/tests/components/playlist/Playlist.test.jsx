import React from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
// import renderer from 'react-test-renderer';
import Adapter from 'enzyme-adapter-react-16';
import { MockedProvider } from '@apollo/react-testing';
import { MemoryRouter, Route } from 'react-router-dom';
import { render } from '@testing-library/react';
import { configure, mount } from 'enzyme';
import configureStore from 'redux-mock-store';
import { act } from 'react-dom/test-utils';

import Playlist from '../../../components/playlist';
import { PLAYLIST_QUERY } from '../../../schemas/playlist';

configure({ adapter: new Adapter() });

// eslint-disable-next-line react/jsx-filename-extension
// jest.mock('react-slick', () => () => (<div id="searchBar" />));

describe('Playlist Component', () => {
    beforeAll(() => {

    });
    const mocks = [
        {
            request: {
                query: PLAYLIST_QUERY,
                variables: { lang: 'EN' },
            },
            result: {
                data: {
                    activities: [
                    ],
                },
            },
        },
    ];

    test('The Playlist renders empty', () => {
        const initialState = {};
        const mockStore = configureStore();
        const store = mockStore(initialState);
        const component = mount(
            <MockedProvider mocks={mocks} addTypename={false}>
                <MemoryRouter>
                    <Playlist store={store} />
                </MemoryRouter>
            </MockedProvider>,
        );
        expect(component).toMatchSnapshot();
    });

    test('no activities defined', () => {
        const initialState = {};
        const mockStore = configureStore();
        const store = mockStore(initialState);
        const { getByText } = render(
            <MockedProvider mocks={mocks} addTypename={false}>
                <MemoryRouter>
                    <Playlist store={store} playlistToRender={[]} />
                </MemoryRouter>
            </MockedProvider>,
        );
        const viewAllText = getByText('View All');
        const noActs = getByText('You currently have no activities in your playlist');
        expect(noActs).toBeDefined();
        expect(viewAllText).toBeDefined();
    });

    test('The playlist renders activity list carousel', () => {
        const initialState = {};
        const mockStore = configureStore();
        const store = mockStore(initialState);
        const mockedPlaylistData = [{
            id: 0,
            user_id: "d4cd70a6-15e0-11ea-8d71-362b9e155667",
            name: "Playlist Name 0",
            activity_list:
                [
                    {
                        id: 3,
                        image: "http://localhost:8001/image4.jpg",
                        type: "video",
                        url: "http://localhost:8001/video1",
                        titles: [
                            {
                                "lang": "en",
                                "title": "ENGLISH Video (3)"
                            }
                        ]
                    }
                ]
        }]
        const { debug, getByText } = render(

            <MemoryRouter>
                <Playlist store={store} session={"d4cd70a6-15e0-11ea-8d71-362b9e155667"} playlistToRender={mockedPlaylistData} />
            </MemoryRouter>

        );
        expect(getByText(mockedPlaylistData[0].name)).toBeDefined();
    });

    test('clicking on the card re-directs you to the activity', () => {
        let Location;
        const initialState = {};
        const mockStore = configureStore();
        const store = mockStore(initialState);
        const mockedPlaylistData = [{
            id: 0,
            user_id: "d4cd70a6-15e0-11ea-8d71-362b9e155667",
            name: "Playlist Name 0",
            activity_list:
                [
                    {
                        id: 3,
                        image: "http://localhost:8001/image4.jpg",
                        type: "video",
                        url: "http://localhost:8001/video1",
                        titles: [
                            {
                                "lang": "en",
                                "title": "ENGLISH Video (3)"
                            }
                        ]
                    }
                ]
        }];
        const { container } = render(
            <MemoryRouter>
                <Playlist store={store} playlistToRender={mockedPlaylistData} />
                <Route
                    path="*"
                    render={({ location }) => { Location = location; return null; }}
                />
            </MemoryRouter>,
        );
        const openPlaylistIcon = container.querySelector('.openPlaylistIcon');
        act(() => {
            openPlaylistIcon.click()
        })
        expect(Location.pathname).toBe(`/view-playlist/${mockedPlaylistData[0].id}`);
    });
});