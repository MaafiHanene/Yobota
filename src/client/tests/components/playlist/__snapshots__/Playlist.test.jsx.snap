// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Playlist Component The Playlist renders empty 1`] = `ReactWrapper {}`;
